zio-query/shared/src/test/scala/zio/query/ZQuerySpec.scala

```diff
@@ -16,6 +16,21 @@ object ZQuerySpec extends ZIOBaseSpec {
           log <- TestConsole.output
         } yield assert(log)(hasSize(equalTo(2)))
       },
+      test("N + 1 on composed queries") {
+        for {
+          ref   <- Ref.make(0)
+          ds     = countingDs(ref)
+          query1 = identityDs.query(IdReq(1)).flatMap(i => ds.query(CountingReq(i)))
+          query2 = identityDs.query(IdReq(2)).flatMap(i => ds.query(CountingReq(i)))
+          res   <- (query1 <~> query2).run
+          i     <- ref.get
+        } yield assertTrue(i == 1, res == ("1", "2"))
+      },
+      test("requests are cached per datasource") {
+        for {
+          res <- identityDs.query(IdReq(1)).flatMap(i => plus1Ds.query(IdReq(i))).run
+        } yield assertTrue(res == 2)
+      },
```

Comment:
This test is irrelevant with the PR. It's something I've been meaning to add for some time now just as a safeguard when making changes to batching logic